jazzmin/templates/admin/edit_inline/tabular.html

```diff
@@ -1,63 +1,46 @@
+                    {% endfor %}
+                    {% if inline_admin_formset.formset.can_delete and inline_admin_formset.has_delete_permission %}<th>{% trans "Delete?" %}</th>{% endif %}
+                </tr></thead>
 
                 <tbody>
                 {% for inline_admin_form in inline_admin_formset %}
                     {% if inline_admin_form.form.non_field_errors %}
-                    <tr>
-                        <td colspan="{{ inline_admin_form|cell_count }}">
-                            {{ inline_admin_form.form.non_field_errors }}
-                        </td>
-                    </tr>
+                        <tr class="row-form-errors"><td colspan="{{ inline_admin_form|cell_count }}">{{ inline_admin_form.form.non_field_errors }}</td></tr>
                     {% endif %}
-                    <tr id="{{ inline_admin_formset.formset.prefix }}-{% if not forloop.last %}{{ forloop.counter0 }}{% else %}empty{% endif %}"{% if forloop.last %} class="empty-form"{% endif %}>
+                    <tr class="form-row {% if inline_admin_form.original or inline_admin_form.show_url %}has_original{% endif %}{% if forloop.last and inline_admin_formset.has_add_permission %} empty-form{% endif %}"
+                        id="{{ inline_admin_formset.formset.prefix }}-{% if not forloop.last %}{{ forloop.counter0 }}{% else %}empty{% endif %}">
                         <td class="original">
                             {% if inline_admin_form.original or inline_admin_form.show_url %}<p>
                                 {% if inline_admin_form.original %}
                                     {{ inline_admin_form.original }}
-                                    {% if inline_admin_form.model_admin.show_change_link and inline_admin_form.model_admin.has_registered_model %}
-                                        <a href="{% url inline_admin_form.model_admin.opts|admin_urlname:'change' inline_admin_form.original.pk|admin_urlquote %}" class="{% if inline_admin_formset.has_change_permission %}inlinechangelink{% else %}inlineviewlink{% endif %}">{% if inline_admin_formset.has_change_permission %}{% trans "Change" %}{% else %}{% trans "View" %}{% endif %}</a>
-                                    {% endif %}
-                                {% endif %}
-                                {% if inline_admin_form.show_url %}
-                                    <a href="{{ inline_admin_form.absolute_url }}">{% trans "View on site" %}</a>
+                                    {% if inline_admin_form.model_admin.show_change_link and inline_admin_form.model_admin.has_registered_model %}<a href="{% url inline_admin_form.model_admin.opts|admin_urlname:'change' inline_admin_form.original.pk|admin_urlquote %}" class="{% if inline_admin_formset.has_change_permission %}inlinechangelink{% else %}inlineviewlink{% endif %}">{% if inline_admin_formset.has_change_permission %}{% trans "Change" %}{% else %}{% trans "View" %}{% endif %}</a>{% endif %}
```

Comment:
Wouldn't it be better to add the icons as opposed to the `Change` and `View` links? It doesn't look too jazzy this way

Reply:
done